packages/frontend/src/components/DashboardTiles/TileForms/AddChartTilesModal.tsx

```diff
@@ -78,49 +97,70 @@ const AddChartTilesModal: FC<Props> = ({ onAddTiles, onClose }) => {
             }),
         );
         onClose();
-    };
+    });
 
     const dashboardTitleName = dashboard?.name
         ? `"${dashboard.name}"`
         : 'dashboard';
 
+    if (!savedCharts || !dashboardTiles || isLoading) return null;
+
     return (
-        <Dialog
-            isOpen={true}
+        <Modal
+            opened={true}
             onClose={onClose}
-            lazy
-            title={`Add charts to ${dashboardTitleName}`}
+            title={
+                <Group spacing="xs">
+                    <MantineIcon
+                        icon={IconChartAreaLine}
+                        size="lg"
+                        color="blue.8"
+                    />
+                    <Title order={4}>
+                        Add saved charts to {dashboardTitleName}
+                    </Title>
+                </Group>
+            }
+            centered
+            withCloseButton={false}
         >
-            <Form
-                name="add_saved_charts_to_dashboard"
-                methods={methods}
-                onSubmit={handleSubmit}
-            >
-                <div className={Classes.DIALOG_BODY}>
+            <Stack spacing="md">
+                <form
+                    id="add-saved-charts-to-dashboard"
+                    onSubmit={handleSubmit}
+                >
                     <MultiSelect
-                        name="savedCharts"
+                        id="saved-charts"
                         label={`Select the charts that you want to add to ${dashboardTitleName}`}
-                        rules={{
-                            required: 'Required field',
-                        }}
-                        items={allSavedCharts}
+                        data={allSavedCharts}
                         disabled={isLoading}
                         defaultValue={[]}
+                        required
+                        withinPortal
+                        itemComponent={SelectItem}
+                        {...form.getInputProps('savedChartsUuids')}
+                        styles={(theme) => ({
+                            input: {
+                                padding: theme.spacing.xxs,
+                            },
```

Comment:
why we need to alter default styles? (im wondering if this is something that should be fixed in a mantine global theme)

Reply:
I just felt like it was a bit cramped, but yeah we should probably fix this in the theme

Reply:
removed this!